integration/generate_config.py

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env python3
```

Comment:
can you add a comment here about what the point of this is (IIUC automate the generation of the file that the integration tests use to run)?  Rather than outputting a file for the integration tests to read, does it make sense for the test to just call a function that does what this file does to generate the set of clients / servers to use?

Reply:
Also, if you leave this strategy of generating the file, can you add this to the pre-commit hooks (to run this file and ensure there are no diffs)? (I think then you'll lose your "TODOs" in the file but I think that's fine and I'd comment those out of the dockerfile instead?)

Reply:
Added a comment and added a github workflow to validate that the configuration in the repo is up to date